### PR TITLE
Refactor test_views_common.py

### PR DESCRIPTION
**Pull request by rebeccacremona**
I set out to convert `test_views_common.py` to pytest format, and to remove its dependency on our legacy JSON test fixtures, swapping in our pytest test fixtures instead.

I did that, but in the process, I decided to split both `views/common.py` and `test_views_common.py` into a few separate modules: one for memento-related views, one for playback, one for views related to our contact and reporting forms... and then whatever was left, still in "common". 

I did this because "common" has never felt descriptive to me, the file was huge, and it was hard to find things in it. I think this is a lot more manageable, and I have wanted to do it for ages.

I made no non-trivial changes to the tests themselves, even if I thought the tests could use improvement, to reduce the scope of this PR. I made _zero_ changes to any of the view code. There is no way you will be able to tell that from the diff 🤣.

As far as reviewing goes... I feel like the only thing worth looking at is, whether you see anything that looks obviously bad/weird in `conftest.py`. I think other than that, it is not worth anybody's time to review: let it be on my head if I messed these tests up during the migration 😂.

### CODE REVIEW

File: perma_web/conftest.py [hunk 1]
```diff
@@ -402,26 +410,46 @@ class InProgressCaptureJobFactory(PendingCaptureJobFactory):
...
+    status = 'success'
+    record_type = 'response'
+    content_type = 'text/html'
+
+
+@register_factory
+class ScreenshotCaptureFactory(CaptureFactory):
+    role = 'screenshot'
+    status = 'success'
+    record_type = 'respsonse'
```
Comment (bensteinberg) [inline perma_web/conftest.py]:
```suggestion
    record_type = 'response'
```

File: perma_web/conftest.py [hunk 2]
```diff
@@ -662,7 +702,119 @@ def f(customer):
...
+            cached_can_play_back=True,
+        ) for time in [today, a_few_days_ago, within_the_last_year, over_a_year_ago, three_years_ago]
+    ]
+
+    return {
+        "domain": domain,
+        "url": url,
+        "links": [
+            {
+                "guid":link.guid,
```
Comment (bensteinberg) [inline perma_web/conftest.py]:
```suggestion
                "guid": link.guid,
```

File: perma_web/conftest.py [hunk 3]
```diff
@@ -662,7 +702,119 @@ def f(customer):
...
+                #
+                "timestamp":  link.creation_timestamp
```
Comment (bensteinberg) [inline perma_web/conftest.py]:
```suggestion
                "timestamp": link.creation_timestamp
```

File: perma_web/conftest.py [hunk 4]
```diff
@@ -662,7 +702,119 @@ def f(customer):
...
+
+### For testing email ###
+
+@pytest.fixture
+def email_details():
+    return {
+        "from_email": 'example@example.com',
+        "custom_subject": 'Just some subject here',
+        "message_text": 'Just some message here.',
+        "refering_page": 'http://elsewhere.com',
```
Comment (bensteinberg) [inline perma_web/conftest.py]: I wish the key here were `referring_page`, but that would require changes elsewhere; maybe I'll make a PR.

**Review by bensteinberg — CHANGES_REQUESTED** [review]
I've made a few suggestions.

File: perma_web/conftest.py (hunk 4 above)
Comment (rebeccacremona) [reply perma_web/conftest.py]: Thanks for these!! 

I can make that change, hold one...

Comment (rebeccacremona) [reply perma_web/conftest.py]: Done! I considered intentionally making a typo in the commit message, like `refferring` 😁

**Review by bensteinberg — APPROVED** [review]
❤️

**Pull request merged**